### PR TITLE
refactor(core,console): remove the local-vm execution path

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Deletes the legacy `node:vm` execution path now that the worker-thread script runner is wired in.

- Remove `packages/core/src/utils/local-vm/` (`runScriptFunctionInLocalVm`).
- Drop every `node:vm` usage outside the script runner worker. `runInThisContext` remains only in `packages/core/src/workers/tasks/script-runner.ts` for filename/stack quality, which the file already documents as not an isolation boundary.
- Drop `getScriptFailureStatusCode` and `buildScriptExecutionErrorBody` from `script-runner/errors.ts`. They existed only to classify a thrown `node:vm` error into a status and an error body; both runners now report failures as values carrying their own `kind`, so the last call sites went away with the legacy path.
- Rename `runScriptInLocalVm` to `runScriptLocally` on `ActionLibrary` and `JwtCustomizerLibrary`, and reword the remaining "local VM" / "sandbox" comments so no future reader mistakes the worker-thread runner for isolation.
- Drop the `action-result-validation` test case that exercised the removed local VM realm; the worker-thread analog already covers the plain object that validation actually receives.
- Console `config.test.ts` evaluates the starter script with `new Function` instead of `node:vm`, keeping the test independent of the Node-only module.

No behavior change: OSS runs already went through the worker pool before this PR.

Closes LOG-13956.

## Testing

Unit tests

> [!NOTE]
> Stacked on #9433; merge that first.

### CODE REVIEW

**Review by simeng-li — APPROVED** [review]

**Review by charIeszhao — APPROVED** [review]

File: packages/core/src/routes/experience/classes/libraries/action-result-validation.test.ts [hunk 1]
```diff
@@ -250,22 +249,9 @@ describe('validatePostSignInActionResult', () => {
...
-    expect(validatePostSignInActionResult({ userId: actionUser.id, result })).toEqual(
-      continueResult
-    );
-  });
-
-  // The worker-pool analog of the separate-realm case above: the value reaches validation after a
-  // structured-clone round trip across the thread boundary.
+  // A script result reaches validation after a structured-clone round trip across the thread
+  // boundary, so it is a plain object from another realm — validation must still treat it as a
+  // no-op.
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/routes/experience/classes/libraries/action-result-validation.test.ts]: Good catch — reworded. The comment now says the value is rebuilt by the structured-clone round trip rather than implying a cross-realm `isPlainObject` edge case.
